x-pack/plugins/security/server/routes/api_keys/update.ts

```diff
@@ -29,51 +34,22 @@ export type UpdateAPIKeyParams =
   | UpdateCrossClusterAPIKeyParams
   | UpdateRestAPIKeyWithKibanaPrivilegesParams;
 
-export type UpdateRestAPIKeyParams = TypeOf<typeof restApiKeySchema>;
-export type UpdateCrossClusterAPIKeyParams = TypeOf<typeof crossClusterApiKeySchema>;
-export type UpdateRestAPIKeyWithKibanaPrivilegesParams = TypeOf<
-  ReturnType<typeof getRestApiKeyWithKibanaPrivilegesSchema>
->;
-
-const restApiKeySchema = schema.object({
-  type: schema.maybe(schema.literal('rest')),
+const updateRestApiKeySchema = restApiKeySchema.extends({
+  name: null,
   id: schema.string(),
-  role_descriptors: schema.recordOf(schema.string(), schema.object({}, { unknowns: 'allow' }), {
-    defaultValue: {},
-  }),
-  metadata: schema.maybe(schema.object({}, { unknowns: 'allow' })),
 });
 
-const crossClusterApiKeySchema = restApiKeySchema.extends({
-  type: schema.literal('cross_cluster'),
-  role_descriptors: null,
-  access: schema.object(
-    {
-      search: schema.maybe(
-        schema.arrayOf(
-          schema.object(
-            {
-              names: schema.arrayOf(schema.string()),
-            },
-            { unknowns: 'allow' }
-          )
-        )
-      ),
-      replication: schema.maybe(
-        schema.arrayOf(
-          schema.object(
-            {
-              names: schema.arrayOf(schema.string()),
-            },
-            { unknowns: 'allow' }
-          )
-        )
-      ),
-    },
-    { unknowns: 'allow' }
-  ),
+const updateCrossClusterApiKeySchema = crossClusterApiKeySchema.extends({
+  name: null,
+  id: schema.string(),
```

Comment:
Rather than keeping two separate schemas for create and update, I opted to reference a common schema and adapt the contents (name vs id).